build fail: binutils@2.33.1 on rhel8 - no makeinfo

`binutils@2.33.1 %gcc@8.3.1` is failing using `spack@develop` (26-Aug-2020, 20d37afafa3be3f9b6c3388e22f813261d4c3fec) on `RHEL8` for `x86_64`

Looks like the dependency on `texinfo` (which provides `makeinfo`) extends to 2.33.1 as well as 2.34. The package.py makes this dependency explicit for 2.34. but not 2.33.1. The build proceeds without problem when `texinfo` is added as an explicit dependency for 2.33.1.

Concrete spec: [binutils-422ehg.yml.txt](https://github.com/spack/spack/files/5130808/binutils-422ehg.yml.txt)
Build log: [binutils-build-out.txt](https://github.com/spack/spack/files/5130801/binutils-build-out.txt)

```
$> spack install -f ./binutils-422ehg.yml
...
==> Installing binutils
==> Using cached archive: /spack/var/spack/cache/_source-cache/archive/0c/0cb4843da15a65a953907c96bad658283f3c4419d6bcc56bf2789db16306adb2.tar.bz2
==> binutils: Executing phase: 'autoreconf'
==> binutils: Executing phase: 'configure'
==> binutils: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

3 warnings found in build log:
     1338    checking target system type... if [ x"-fpic" != x ]; then \
     1339      /spack/lib/spack/env/gcc/gcc -c -DHAVE_CONFIG_H -g -O2  -I. -I./../include  -W -Wall -Wwrite-strings -Wc++-compat -Wstrict-prototypes -Wshadow=local -pedantic
             -D_GNU_SOURCE -fpic ./fibheap.c -o pic/fibheap.o; \
     1340    else true; fi
     1341    x86_64-pc-linux-gnu
     1342    checking for x86_64-pc-linux-gnu-gcc... /spack/lib/spack/env/gcc/gcc
     1343    ./dyn-string.c: In function 'dyn_string_insert_cstr':
  >> 1344    ./dyn-string.c:280:3: warning: 'strncpy' output truncated before terminating nul copying as many bytes from a string as its length [-Wstringop-truncation]
     1345       strncpy (dest->s + pos, src, length);
     1346       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     1347    ./dyn-string.c:272:16: note: length computed here
     1348       int length = strlen (src);
     1349                    ^~~~~~~~~~~~
     1350    if [ x"" != x ]; then \

     ...

     1390    else true; fi
     1391    /spack/lib/spack/env/gcc/gcc -c -DHAVE_CONFIG_H -g -O2  -I. -I./../include  -W -Wall -Wwrite-strings -Wc++-compat -Wstrict-prototypes -Wshadow=local -pedantic  -D
             _GNU_SOURCE ./cp-demangle.c -o cp-demangle.o
     1392    if [ x"-fpic" != x ]; then \
     1393      /spack/lib/spack/env/gcc/gcc -c -DHAVE_CONFIG_H -g -O2  -I. -I./../include  -W -Wall -Wwrite-strings -Wc++-compat -Wstrict-prototypes -Wshadow=local -pedantic
             -D_GNU_SOURCE -fpic ./getruntime.c -o pic/getruntime.o; \
     1394    else true; fi
     1395    ./dyn-string.c: In function 'dyn_string_insert_cstr':
  >> 1396    ./dyn-string.c:280:3: warning: 'strncpy' output truncated before terminating nul copying as many bytes from a string as its length [-Wstringop-truncation]
     1397       strncpy (dest->s + pos, src, length);
     1398       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     1399    ./dyn-string.c:272:16: note: length computed here
     1400       int length = strlen (src);
     1401                    ^~~~~~~~~~~~
     1402    yes

     ...

     2229    else \
     2230      rc=$?; \
     2231      $restore $backupdir/* `echo "./bfd.info" | sed 's|[^/]*$||'`; \
     2232    fi; \
     2233    rm -rf $backupdir; exit $rc
     2234    /tmp/root/spack-stage/spack-stage-binutils-2.33.1-422ehgikkmt4cghrtx5bracwl34otohy/spack-src/missing: line 81: makeinfo: command not found
  >> 2235    WARNING: 'makeinfo' is missing on your system.
     2236             You should only need it if you modified a '.texi' file, or
     2237             any other file indirectly affecting the aspect of the manual.
     2238             You might want to install the Texinfo package:
     2239             <http://www.gnu.org/software/texinfo/>
     2240             The spurious makeinfo call might also be the consequence of
     2241             using a buggy 'make' (AIX, DU, IRIX), in which case you might

See build log for details:
  /tmp/root/spack-stage/spack-stage-binutils-2.33.1-422ehgikkmt4cghrtx5bracwl34otohy/spack-build-out.txt
```

@michaelkuhn @mathsen @mwkrentel

--- a/var/spack/repos/builtin/packages/binutils/package.py
+++ b/var/spack/repos/builtin/packages/binutils/package.py
@@ -44,6 +44,7 @@ class Binutils(AutotoolsPackage, GNUMirrorPackage):
     patch('update_symbol-2.26.patch', when='@2.26')
 
     depends_on('zlib')
+    depends_on('diffutils', type='build')
     depends_on('gettext', when='+nls')
 
     # Prior to 2.30, gold did not distribute the generated files and

